server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -646,7 +646,12 @@
 
         Ternary<Long, Boolean, ListProjectResourcesCriteria> domainIdRecursiveListProject = new Ternary<Long, Boolean, ListProjectResourcesCriteria>(cmd.getDomainId(), cmd.isRecursive(), null);
 
-        _accountMgr.buildACLSearchParameters(caller, null, cmd.getAccountName(), cmd.getProjectId(), permittedAccounts, domainIdRecursiveListProject, listAll, false);
+        Long projectId = cmd.getProjectId();
+        if (resourceType.equalsIgnoreCase("project") && projectId == null) {
+            projectId = Long.parseLong(resourceId);
```

Comment:
@DaanHoogland 
resourceId might be a uuid
it is better to have check like 
https://github.com/apache/cloudstack/blob/master/server/src/main/java/com/cloud/api/query/QueryManagerImpl.java#L681

Reply:
yes @ustcweizhou but the search parameter is always Long, so we'll have to either refuse it if it is a uuid or add a different search. right now an exception is thrown.
I am thinkin g of abandoning this and just fix the test by setting the projectId in marvin.

Reply:
implemented a fetch of the id based on assumed uuid